### PR TITLE
feat(handlers): add getEmoji() & getEmojis()

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**
Adds getEmoji functions.
Fixes emoji update event not caching emojis.

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/api/handlers/guild.ts [hunk 1]
```diff
@@ -304,6 +305,52 @@ export function emojiURL(id: string, animated = false) {
   return `https://cdn.discordapp.com/emojis/${id}.${animated ? "gif" : "png"}`;
 }
 
+/**
+ * ⚠️ **If you need this, you are probably doing something wrong. Always use cache.guilds.get()?.emojis
+ * 
+ * Returns a list of emojis for the given guild.
+ */
+export async function getEmojis(guildID: string, addToCache = false) {
```
Comment (ayntee) [inline src/api/handlers/guild.ts:313]: The default value for `addToCache` should be `true`.

File: src/api/handlers/guild.ts [hunk 2]
```diff
@@ -304,6 +305,52 @@ export function emojiURL(id: string, animated = false) {
...
+
+/**
+ * ⚠️ **If you need this, you are probably doing something wrong. Always use cache.guilds.get()?.emojis
+ * 
+ * Returns an emoji for the given guild and emoji ID.
+ */
+export async function getEmoji(
+  guildID: string,
+  emojiID: string,
+  addToCache = false,
```
Comment (ayntee) [inline src/api/handlers/guild.ts:335]: The default value for `addToCache` should be `true`.

Comment (ayntee) [inline src/api/handlers/guild.ts:335]:
```suggestion
  addToCache = true,
```

File: src/api/handlers/guild.ts (hunk 1 above)
Comment (ayntee) [inline src/api/handlers/guild.ts:313]:
```suggestion
export async function getEmojis(guildID: string, addToCache = true) {
```

File: src/api/controllers/guilds.ts [hunk 3]
```diff
@@ -114,6 +114,8 @@ export async function handleInternalGuildEmojisUpdate(data: DiscordPayload) {
   const cachedEmojis = guild.emojis;
   guild.emojis = payload.emojis;
 
+  cacheHandlers.set("guilds", payload.guild_id, guild);
```
Comment (ayntee) [inline src/api/controllers/guilds.ts:117]: This is an irrelevant change, remove this.

File: src/api/controllers/guilds.ts [hunk 4]
```diff
@@ -114,6 +114,8 @@ export async function handleInternalGuildEmojisUpdate(data: DiscordPayload) {
...
+
```
Comment (ayntee) [inline src/api/controllers/guilds.ts:118]: This is an irrelevant change, remove this.

File: src/api/handlers/guild.ts [hunk 5]
```diff
@@ -304,6 +305,52 @@ export function emojiURL(id: string, animated = false) {
   return `https://cdn.discordapp.com/emojis/${id}.${animated ? "gif" : "png"}`;
 }
 
+/**
+ * ⚠️ **If you need this, you are probably doing something wrong. Always use cache.guilds.get()?.emojis
```
Comment (ayntee) [inline src/api/handlers/guild.ts:309]: Move this at the end of the comment block, the description of the function should come first.

File: src/api/handlers/guild.ts [hunk 6]
```diff
@@ -304,6 +305,52 @@ export function emojiURL(id: string, animated = false) {
...
+    const guild = await cacheHandlers.get("guilds", guildID);
+    if (!guild) throw new Error(Errors.GUILD_NOT_FOUND);
+    cacheHandlers.set("guilds", guildID, { ...guild, emojis: result });
+  }
+
+  return result;
+}
+
+/**
+ * ⚠️ **If you need this, you are probably doing something wrong. Always use cache.guilds.get()?.emojis
```
Comment (ayntee) [inline src/api/handlers/guild.ts:328]: Move this at the end of the comment block, the description of the function should come first.

**Review by itohatweb — COMMENTED** [review]
I'll make separate PR for that

File: src/api/handlers/guild.ts [hunk 7]
```diff
@@ -304,6 +305,52 @@ export function emojiURL(id: string, animated = false) {
...
+
+  if (addToCache) {
+    const guild = await cacheHandlers.get("guilds", guildID);
+    if (!guild) throw new Error(Errors.GUILD_NOT_FOUND);
+    cacheHandlers.set(
+      "guilds",
+      guildID,
+      { ...guild, emojis: [...guild.emojis, result] },
+    );
+  }
```
Comment (ayntee) [inline src/api/handlers/guild.ts:349]:
```suggestion
    const guild = await cacheHandlers.get("guilds", guildID);
    if (!guild) throw new Error(Errors.GUILD_NOT_FOUND);
    guild.emojis.push(result);
    cacheHandlers.set(
      "guilds",
      guildID,
      guild,
    );
  }
```

File: src/api/handlers/guild.ts [hunk 8]
```diff
@@ -304,6 +305,53 @@ export function emojiURL(id: string, animated = false) {
...
+export async function getEmojis(guildID: string, addToCache = true) {
+  const result = await RequestManager.get(
+    endpoints.GUILD_EMOJIS(guildID),
+  ) as Emoji[];
+
+  if (addToCache) {
+    const guild = await cacheHandlers.get("guilds", guildID);
+    if (!guild) throw new Error(Errors.GUILD_NOT_FOUND);
+    cacheHandlers.set("guilds", guildID, { ...guild, emojis: result });
+  }
```
Comment (ayntee) [inline src/api/handlers/guild.ts:322]:
```suggestion
  if (addToCache) {
    const guild = await cacheHandlers.get("guilds", guildID);
    if (!guild) throw new Error(Errors.GUILD_NOT_FOUND);
    guild.emojis = result;
    cacheHandlers.set("guilds", guildID, guild);
  }
```

**Review by ayntee — APPROVED** [review]
Thank you!

**Pull request merged**